### PR TITLE
dtv-scan-tables: update and migrate to by-name and finalAttrs

### PR DESCRIPTION
**Pull request by Yarny0**
## Description of changes

* update to newest release 2024-03-24
* move package to `pkgs/by-name`
* replace `rec` with `finalAttrs` pattern
* push `with lib;` down into `meta`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

## Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>dtv-scan-tables</li>
    <li>tvheadend</li>
    <li>tvheadend.man</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Misterio77 — APPROVED** [review]
~Might be a good idea to split each change (update, by-name, finalAttrs) into a separate commit. But just a nit, otherwise~ LGTM!

**Comment by Misterio77** [comment]
Nevermind, just saw it's already split haha

**Comment by Misterio77** [comment]
Result of `nixpkgs-review pr 309331` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>dtv-scan-tables</li>
    <li>tvheadend</li>
    <li>tvheadend.man</li>
  </ul>
</details>

File: pkgs/by-name/dt/dtv-scan-tables/package.nix [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "dtv-scan-tables";
-  version = "2022-04-30-57ed29822750";
+  version = "2024-03-24-7098bdd27548";
 
   src = fetchurl {
-    url = "https://linuxtv.org/downloads/${pname}/${pname}-${version}.tar.bz2";
-    hash = "sha256-amJoqjkkWTePo6E5IvwBWj+mP/gi9LDWTTPXE1Cm7J4=";
+    url = "https://linuxtv.org/downloads/${finalAttrs.pname}/${finalAttrs.pname}-${finalAttrs.version}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/by-name/dt/dtv-scan-tables/package.nix:12]:
```suggestion
    url = "https://linuxtv.org/downloads/dtv-scan-tables/dtv-scan-tables-${finalAttrs.version}.tar.bz2";
```
The url will always be wrong unless the string matches this exactly, which makes substituting it kinda pointless

Comment (Yarny0) [reply pkgs/by-name/dt/dtv-scan-tables/package.nix]: Done, thanks.  And rebased onto current `nixos-unstable`.

**Pull request merged**